lib/Sema/CSDiagnostics.cpp

```diff
@@ -2278,6 +2278,22 @@ void ContextualFailure::tryFixIts(InFlightDiagnostic &diagnostic) const {
     return;
 }
 
+static InFlightDiagnostic emitMissingNullaryCallDiagnostic(ConstraintSystem &cs,
+                                                           FunctionType *srcFT,
+                                                           Expr *anchor) {
+  auto &diags = cs.getASTContext().Diags;
+  if (auto declRefExpr = dyn_cast<DeclRefExpr>(anchor)) {
+    if (auto decl = declRefExpr->getDecl())
+      return diags.diagnose(anchor->getLoc(), diag::missing_nullary_call_decl,
+                            srcFT, decl->getDescriptiveKind(),
+                            decl->getBaseName(), srcFT->getResult(),
+                            !srcFT->getParams().empty());
+  }
+
+  return diags.diagnose(anchor->getLoc(), diag::missing_nullary_call, srcFT,
```

Comment:
Could you please make it a method on ContextualFailure and use `emitDiagnostic` instead of `DiagnosticEngine` directly? I'm trying to change the way diagnostics are emitted in the future - https://github.com/apple/swift/pull/30967

Reply:
I think we'll need to duplicate it in `MissingCallFailure` too since this error can be emitted from both. It's fairly short though, so probably worth it to take advantage of the upcoming changes

Reply:
Could we maybe inherit `MissingCallFailure` from `ContextualFailure`?

Reply:
Maybe, I'll look into doing that